demos/multi_channel/object_detection_demo_yolov3/main.cpp

```diff
@@ -214,8 +212,16 @@ void drawDetections(cv::Mat& img, const std::vector<DetectionObject> detections)
                                   (float)f.ymin,
                                   (float)(f.xmax-f.xmin),
                                   (float)(f.ymax-f.ymin)), 
-                      cv::Scalar(255, 0, 0), 
+                      colors[(int)f.class_id], 
                       2);
+        cv::putText(img, 
+                    coco_labels[(int)f.class_id], 
+                    cv::Point((float)f.xmin,
+                              (float)f.ymin),
+                    cv::FONT_HERSHEY_COMPLEX,
+                    1,
+                    colors[(int)f.class_id],
```

Comment:
You need to handle the possibility of `f.class_id >= colors.size()` (which can happen due to user error, or unexpected model behavior).